public/modules/custom/grants_attachments/src/Controller/GrantsAttachmentsController.php

```diff
@@ -150,20 +92,20 @@ public function deleteAttachment(string $submission_id, string $integration_id):
     }
     catch (\Exception $e) {
       $this->messenger()->addError($e->getMessage());
-      return new RedirectResponse($this->request->getMainRequest()->get('destination'));
+      return new RedirectResponse($this->requestStack->getMainRequest()->get('destination'));
```

Comment:
This could be refactored to use `redirect.destination` service: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Routing%21RedirectDestination.php/class/RedirectDestination/10